src/core/value/constant.jl

```diff
@@ -123,13 +123,23 @@ end
 identify(::Type{Value}, ::Val{API.LLVMConstantArrayValueKind}) = ConstantArray
 identify(::Type{Value}, ::Val{API.LLVMConstantDataArrayValueKind}) = ConstantArray
 
-ConstantArray(typ::LLVMType, data::Vector{T}) where {T<:Constant} =
-    ConstantArray(API.LLVMConstArray(typ, data, length(data)))
-ConstantArray(typ::IntegerType, data::Vector{T}) where {T<:Integer} =
-    ConstantArray(typ, map(x->ConstantInt(convert(T,x),context(typ)), data))
-ConstantArray(typ::FloatingPointType, data::Vector{T}) where {T<:AbstractFloat} =
-    ConstantArray(typ, map(x->ConstantFP(convert(T,x),context(typ)), data))
+function ConstantArray(typ::LLVMType, data::AbstractArray{T,N}) where {T<:Constant,N}
+    if N == 1
+        return ConstantArray(API.LLVMConstArray(typ, Array(data), length(data)))
+    end
+
+    # FIXME: once we drop support for julia 1.0, we can use `eachslice(data, dims=1)` instead of this monstrosity
```

Comment:
Maybe have both, guarded by a check against VERSION. That way bots like FemtoCleaner.jl can automatically remove the old crufty code once we bump our Julia compatibility.